app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments2/details/PaymentDetailsDestination.kt

```diff
@@ -243,17 +270,46 @@ internal fun MemberChargeDetailsScreen(
   }
 }
 
+internal fun getNextCharge(memberCharge: MemberCharge, paymentOverview: PaymentOverview): MemberCharge? {
+  val index = (paymentOverview.pastCharges?.indexOf(memberCharge) ?: 0) - 1
+  return if (index < 0 && paymentOverview.memberCharge != null) {
+    paymentOverview.memberCharge
+  } else if (index >= 0) {
+    paymentOverview.pastCharges?.get(index)
+  } else {
+    null
+  }
+}
```

Comment:
Do you think this piece of logic could live in the ViewModel instead? And the UI can just show the result on the screen directly without having to do these calculations itself?

Reply:
Good idea. I will fix it in the PRs above to avoid too many conflicts